bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/AnnotationUtils.java

```diff
@@ -43,12 +43,17 @@
      * @param globalAnnotMap The global annotation map
      * @param bType          The type for which annotations need to be set
      */
-    public static void processAnnotations(MapValue globalAnnotMap, Type bType) {
+    public static void processAnnotations(MapValue globalAnnotMap, Type bType, Strand strand) {
         if (!(bType instanceof BAnnotatableType)) {
             return;
         }
 
         BAnnotatableType type = (BAnnotatableType) bType;
+        if (type.getTag() == TypeTags.SERVICE_TAG) {
+            processServiceAnnotations(globalAnnotMap, (BServiceType) bType, strand);
```

Comment:
`processServiceAnnotations` used to be codegen-ed separately for services to specifically recompute the annotations in services that are created within a function (via https://github.com/ballerina-platform/ballerina-lang/blob/v2.0.0-Preview7/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmInstructionGen.java#L1625).

To continue to support the same, IMO we need to do something similar.

The second annotation access in the following sample used to return `{i: 202}`, but now returns `{i: 101}`.

```ballerina
annotation record {int i;} x on service;

int globalInt = 101;

public function getService() returns service object {} {
    return @x {i: globalInt} service object {
        
    };
}

public function main() {
    typedesc<service object {}> t = typeof getService();
    var v = t.@x; // {"i":101}

    globalInt = 202;
    t = typeof getService();
    v = t.@x; // used to be {"i":202}, now {"i":101}
}
```

Will we be fixing this anytime soon given that it's a breaking change?

Reply:
Could this be this issue, I mean this behavior https://github.com/ballerina-platform/ballerina-lang/issues/26838

Reply:
Issue to re-visit this annot impl: https://github.com/ballerina-platform/ballerina-lang/issues/27348

I think we will have to consider object constructors for the above issue.